crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1094,6 +1115,16 @@ where
         original_tx_digest: &TransactionDigest,
         state: ProcessTransactionState,
     ) -> AggregatorProcessTransactionError {
+        let quorum_threshold = self.committee.quorum_threshold();
+
+        // Return system overload error if we see >= 2f + 1 overloaded stake.
+        if state.overloaded_stake >= quorum_threshold {
```

Comment:
nit: based on your comment above, it might make sense then to move this to the `if !state.retryable` block below?